src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,253 @@
+import com.google.api.codegen.viewmodel.SettingsDocView;
+import com.google.api.codegen.viewmodel.StaticLangApiMethodView;
+import com.google.api.codegen.viewmodel.StaticLangXApiView;
+import com.google.api.codegen.viewmodel.StaticLangXCombinedView;
+import com.google.api.codegen.viewmodel.StaticLangXSettingsView;
+import com.google.api.codegen.viewmodel.ViewModel;
+import com.google.api.gax.core.RetrySettings;
+import com.google.api.tools.framework.model.Field;
+import com.google.api.tools.framework.model.Interface;
+import com.google.api.tools.framework.model.Method;
+import com.google.api.tools.framework.model.Model;
+import com.google.api.tools.framework.model.ProtoElement;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableSet;
+
+import io.grpc.Status.Code;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map.Entry;
+
+public class GoGapicSurfaceTransformer implements ModelToViewTransformer {
+
+  private static final String XAPI_TEMPLATE_FILENAME = "go/main.snip";
+  private static final String SAMPLE_TEMPLATE_FILENAME = "go/example.snip";
+  private static final String DOC_TEMPLATE_FILENAME = "go/doc.snip";
+
+  private final GapicCodePathMapper pathMapper;
+  private final ApiCallableTransformer apiCallableTransformer = new ApiCallableTransformer();
+  private final ApiMethodTransformer apiMethodTransformer = new ApiMethodTransformer();
+  private final PageStreamingTransformer pageStreamingTransformer = new PageStreamingTransformer();
+  private final PathTemplateTransformer pathTemplateTransformer = new PathTemplateTransformer();
+  private final ServiceTransformer serviceTransformer = new ServiceTransformer();
+
+  public GoGapicSurfaceTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME);
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> models = new ArrayList<ViewModel>();
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      models.add(generate(model, service, apiConfig));
+    }
+    return models;
+  }
+
+  private StaticLangXCombinedView generate(Model model, Interface service, ApiConfig apiConfig) {
+    StaticLangXCombinedView.Builder view = StaticLangXCombinedView.newBuilder();
+
+    String packageName = getPackageName(apiConfig.getPackageName());
+    view.packageName(packageName);
+
+    view.outputPath(
+        apiConfig.getPackageName().replace('/', File.separatorChar)
+            + File.separator
+            + getReducedServiceName(service)
+            + "_client.go");
+
+    view.templateFileName(XAPI_TEMPLATE_FILENAME);
+    view.name(getClientPrefix(model, service, packageName));
+    view.serviceName(service.getSimpleName());
```

Comment:
Don't use strings from the model directly, go through SurfaceNamer, even in cases where the string isn't modified. This provides a consistent flow, and a reliable way to intercept strings in the flow if they do need a transformation later.